src/Client/Update/SpreadsheetUpdate.fs

```diff
@@ -97,6 +97,17 @@ module Spreadsheet =
                     | IsTable -> Controller.BuildingBlocks.addDataAnnotation data state
                     | IsMetadata -> failwith "Unable to add data annotation in metadata view"
                 nextState, model, Cmd.none
+            | AddTemplate table ->               
+                let index = Some (Spreadsheet.BuildingBlocks.Controller.SidebarControllerAux.getNextColumnIndex model.SpreadsheetModel)
+                /// Filter out existing building blocks and keep input/output values.
+                let options = Some ARCtrl.TableJoinOptions.WithValues // If changed to anything else we need different logic to keep input/output values
+                let msg t i o = JoinTable(t, i, o)
+                let nextState =
+                    let _ =
```

Comment:
isnt this missing the prepare template logic?

Reply:
Do previously existing inputs/output persist after adding template in standalone swate?

Reply:
It is already working fine and I do not know what shalle be prepared because everything is coming together and there is not enough code in the joinTable function to spread it into two
![grafik](https://github.com/user-attachments/assets/c38e9c81-afed-4a6b-90a0-b73abf2f5053)

Reply:
And yes, values are taken over when the template is added